client/src/app/shared/validators/one-of-validator.ts

```diff
@@ -0,0 +1,30 @@
+import { ValidationErrors, AbstractControl } from '@angular/forms';
+
+/**
+ * Custom validator class
+ */
+export class OneOfValidator {
+    /**
+     * Function to confirm that at least one of the given form controls has a valid value.
+     *
+     * @param keys Undefined amount of form control names.
+     *
+     * @returns An error if all of these form controls have no valid values or null if at least one is filled.
+     */
+    public static validation(...keys: string[]): ValidationErrors | null {
+        return (control: AbstractControl): ValidationErrors | null => {
+            const formControls = [];
```

Comment:
`... = keys.map(key => control.get(key))`. Is it guaranteed, that every of these controls exist?